UI/MapWnd.cpp

```diff
@@ -1915,7 +1915,7 @@ void MapWnd::RenderMovementLine(const MapWnd::MovementLineData& move_line, float
 
     // set vertex positions to outline a quad for each move line vertex
     for (std::vector<MovementLineData::Vertex>::const_iterator verts_it = vertices.begin();
-         verts_it != vertices.end(); ++verts_it)
+         verts_it != --vertices.end(); ++verts_it)
```

Comment:
Later on we do

```cpp
if (dots_added_to_buffer >= BUFFER_CAPACITY)
            break; // can't fit any more!

        // get next two vertices
        const MovementLineData::Vertex& vert1 = *verts_it;
        verts_it++;
        const MovementLineData::Vertex& vert2 = *verts_it;
```

verts_it can become end() which is adjacent memory not belonging to the vector. --end() ensures we are one off from the edge of the vector.